### PR TITLE
Don't test revdeps if main build fails

### PR DESCRIPTION
**Pull request by talex5**

### CODE REVIEW

File: service/pipeline.ml [hunk 1]
```diff
@@ -130,7 +131,7 @@ let build_with_cluster ~ocluster ~analysis ~master source =
         and+ build = Node.of_job `Built image ~label:"build"
         and+ tests = Node.of_job `Built tests ~label:"tests"
```
Comment (kit-ty-kate) [inline service/pipeline.ml:132]: the same thing is required for the tests I believe

Comment (kit-ty-kate) [inline service/pipeline.ml:132]: the same thing is required for the tests I believe

edit: oops sorry for the double post

Comment (talex5) [reply service/pipeline.ml:132]: Tests should be covered already by the `after:image` bit:
```ocaml
        let tests =
          let spec = test_spec ~platform pkg ~after:image in
          Build.v ocluster ~label:"test" ~base ~spec ~master source
```

Comment (kit-ty-kate) [reply service/pipeline.ml:132]: Oh I see! Sorry for the disturbance.

**Review by kit-ty-kate — APPROVED** [review]

**Pull request merged**